compiler/src/dotty/tools/dotc/parsing/Parsers.scala

```diff
@@ -1447,8 +1447,11 @@ object Parsers {
       }
     }
 
-    private def makeKindProjectorTypeDef(name: TypeName): TypeDef =
-      TypeDef(name, WildcardTypeBoundsTree()).withFlags(Param)
+    private def makeKindProjectorTypeDef(name: TypeName): TypeDef = {
+      val isVarianceAnnotated = name.lastPart.startsWith("+") || name.lastPart.startsWith("-")
+      val unannotatedName = if (isVarianceAnnotated) name.mapLast(_.drop(1)) else name
```

Comment:
Would be nice to add a comment mentioning that we can strip the variance because it's inferred.

Reply:
@smarter Added a comment explaining the decision to just strip the variance